src/Components/ProductListingPage/ProductsSection/ProductsCard/ProductsCard.jsx

```diff
@@ -1,32 +1,47 @@
-import { useCart, useWishlist } from "../../../../Context";
+import { useAuth, useCart, useModal, useWishlist } from "../../../../Context";
 import { Link } from "react-router-dom";
 
 const ProductsCard = ({ item }) => {
   const { title, price, rating, totalRating, src1, newestArrival, inStock } =
     item;
   const { cartState, cartDispatch } = useCart();
   const { wishlist, setWishlist } = useWishlist();
+  const { auth } = useAuth();
+  const { setShowLoginModal } = useModal();
 
   const addCartClick = () => {
-    cartDispatch({ type: "addToCart", payload: item });
+    auth.login
+      ? cartDispatch({ type: "addToCart", payload: item })
+      : setShowLoginModal(true);
   };
 
   const removeFromCart = () => {
-    cartDispatch({ type: "removeFromCart", payload: item });
+    auth.login && cartDispatch({ type: "removeFromCart", payload: item });
   };
 
   const addWishlistClick = () => {
-    setWishlist((oldCart) => {
-      return [...new Set([...oldCart, item])];
-    });
+    auth.login
+      ? setWishlist((oldCart) => {
+          return [...new Set([...oldCart, item])];
+        })
+      : setShowLoginModal(true);
   };
 
   const removeFromWishlist = () => {
-    setWishlist((oldWishlist) => {
-      return oldWishlist.filter((el) => {
-        return el._id !== item._id;
+    auth.login &&
+      setWishlist((oldWishlist) => {
+        return oldWishlist.filter((el) => {
+          return el._id !== item._id;
+        });
       });
-    });
+  };
+
+  const cartButtonStatus = () => {
+    cartState.cart.includes(item) ? removeFromCart() : addCartClick();
```

Comment:
When clicking add to card CTA of same product twice, you can either show an alert saying already added to cart or show an option to increment quantity so that we do not leave the user guessing about what's happening.

Reply:
Fixed